asterius/src/Asterius/Passes/SafeCCall.hs

```diff
@@ -158,36 +159,76 @@ splitSingleBlock vts save_instrs base_k base_block@RelooperBlock {..} =
             this_preblock_instrs = bsPreBlockInstrs s
             save_target_instr =
               SetLocal {index = 2, value = ConstI32 $ fromIntegral next_id}
-            reset_instr = CallImport
-              { target' = "__asterius_resetPromise",
-                operands = [],
-                callImportReturnTypes = []
-                }
+
+
+            reset_instrs =
+               -- Reset:
+               --  * saved local regs
+               --  * ffi_func
+               --  * ffi_return
+               --
+               [ Store
+                   { bytes  = 8
+                   , offset = fromIntegral offset_StgTSO_ffi_func
+                   , ptr = Unary
+                       { unaryOp = WrapInt64
+                       , operand0 = getLVal currentTSO
+                       }
+                   , value = ConstI64 0
+                   , valueType = I64
+                   }
+               ]
+               ++ [ Store
+                   { bytes  = 8
+                   , offset = fromIntegral offset_StgTSO_ffi_return
+                   , ptr = Unary
+                       { unaryOp = WrapInt64
+                       , operand0 = getLVal currentTSO
+                       }
+                   , value = ConstI64 0
+                   , valueType = I64
+                   }
+               ]
+               ++ [ Store
```

Comment:
I don't really understand why we need to reset to 0 all the saved regs after an FFI call, but if we don't bad things happen. Avoiding it or only reseting some of them would generate less code.

Reply:
We can call `memset` for smaller & faster reset code. The point of resetting is: a function with async imports may be entered multiple times, and we must ensure no state from previous async import calls is leaked.